clients/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/NessieUtils.scala

```diff
@@ -246,7 +247,16 @@ object NessieUtils {
   ): Reference = {
     val refName = getCurrentRefName(currentCatalog, catalog)
```

Comment:
Why not update the existing getCurrentRefName to return a tuple of name + optional-hash?

Reply:
originally thought of changing but didn't change as I had to explore some Scala syntaxes if I change.

Updated now.